research/ag_news/dynamic_layer_exchange/README.md

```diff
@@ -0,0 +1,74 @@
+# Dynamic Layer Exchange Federated Learning For Fine-tuning BERT
+This example leverages the dynamic layer exchanger, where instead of exchanging the entire model between the server and clients in every training round,
+only certain tensors are exchanged, and weighted averaging is performed in a per-tensor fashion on the server side.
+
+In each training round, the tensors to be exchanged with the server are selected based their "drift norm" in that training round, which is defined as the l2-norm of the difference between their values at the end of that training round and their initial values at the beginning of the same training round.
+
+You can customize how tensors are selected and how many tensors to exchange, but more on this later.
+
+This example fine-tunes a pre-trained BERT encoder to perform text classification on the AG News dataset in a federated setting.
+Each client performs Dirichlet sampling on its training set so we get heterogeneous data distributions across clients.
+
+## Running the Example
+In order to run the example, first ensure you have [installed the dependencies in your virtual environment according to the main README](/README.md#development-requirements) and it has been activated.
+
+## Information about Dataset
+client_data.py handles all data preprosessiong steps and is responsible for constructing the dataloaders for training and evaluation.
```

Comment:
The spelling of preprocessing appears to have gone a little sideways here 🙂

Reply:
😂